lib/ecto/query.ex

```diff
@@ -2667,6 +2667,34 @@ defmodule Ecto.Query do
                  where: l.inserted_at > c.updated_at,
                  preload: [:author, comments: {c, likes: l}]
 
+  ## Choosing Between Preloading with Joins vs. Separate Queries
```

Comment:
I wasn't sure whether to make it a separate section with a heading or not, but I ended up doing it because it got too big to just be part of the normal text.

I also put it "first" as it seemed most relevant to the preceeding setion flow-wise. Happy to re-evaluate and change!